To: vim_dev@googlegroups.com
Subject: Patch 7.3.106
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.106
Problem:    When 'cursorbind' is set another window may scroll unexpectedly
            when 'scrollbind' is also set. (Xavier Wang)
Solution:   Don't call update_topline() if 'scrollbind' is set.
Files:      src/move.c


*** ../vim-7.3.105/src/move.c	2010-08-15 21:57:29.000000000 +0200
--- src/move.c	2011-01-22 20:58:50.000000000 +0100
***************
*** 2882,2890 ****
  	    if (has_mbyte)
  		mb_adjust_cursor();
  # endif
- 
  	    redraw_later(VALID);
! 	    update_topline();
  # ifdef FEAT_WINDOWS
  	    curwin->w_redr_status = TRUE;
  # endif
--- 2882,2892 ----
  	    if (has_mbyte)
  		mb_adjust_cursor();
  # endif
  	    redraw_later(VALID);
! 
! 	    /* Only scroll when 'scrollbind' hasn't done this. */
! 	    if (!curwin->w_p_scb)
! 		update_topline();
  # ifdef FEAT_WINDOWS
  	    curwin->w_redr_status = TRUE;
  # endif
*** ../vim-7.3.105/src/version.c	2011-01-22 01:13:33.000000000 +0100
--- src/version.c	2011-01-22 21:03:21.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     106,
  /**/

-- 
Never eat yellow snow.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
